### PR TITLE
Cosmosdb

### PR DESCRIPTION
**Pull request by JoshLove-msft**
Fixes https://github.com/Azure/azure-sdk-for-net/issues/41978

### CODE REVIEW

File: sdk/provisioning/Azure.Provisioning/src/cosmosdb/CosmosDBAccount.cs [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        /// <param name="consistencyPolicy"></param>
+        /// <param name="accountOfferType"></param>
+        /// <param name="accountLocations"></param>
+        /// <param name="parent"></param>
+        /// <param name="name"></param>
+        /// <param name="version"></param>
+        /// <param name="location"></param>
+        public CosmosDBAccount(
+            IConstruct scope,
+            CosmosDBAccountKind? kind = default,
```
Comment (m-nash) [inline sdk/provisioning/Azure.Provisioning/src/cosmosdb/CosmosDBAccount.cs:33]: In general if these items are settable on the Properties I think we should not have them in the constructor as optional as it makes the ctor very large in some cases.

File: sdk/provisioning/Azure.Provisioning/src/cosmosdb/CosmosDBKey.cs [hunk 2]
```diff
@@ -0,0 +1,56 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.ComponentModel;
+
+namespace Azure.Provisioning.CosmosDB
+{
+    /// <summary> Role definition. </summary>
+    public readonly partial struct CosmosDBKey : IEquatable<CosmosDBKey>
```
[the remark this answers is not in the document]
Comment (JoshLove-msft) [reply sdk/provisioning/Azure.Provisioning/src/cosmosdb/CosmosDBKey.cs:10]: Nope I couldn't find it.

File: sdk/provisioning/Azure.Provisioning/src/cosmosdb/CosmosDBAccount.cs (hunk 1 above)
Comment (JoshLove-msft) [reply sdk/provisioning/Azure.Provisioning/src/cosmosdb/CosmosDBAccount.cs:33]: kind seems pretty fundamental though, but I will keep that in mind.

File: sdk/provisioning/Azure.Provisioning/src/cosmosdb/CosmosDBKey.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (JoshLove-msft) [reply sdk/provisioning/Azure.Provisioning/src/cosmosdb/CosmosDBKey.cs:10]: I could only find this output type - https://learn.microsoft.com/en-us/dotnet/api/microsoft.azure.management.cosmosdb.models.databaseaccountkeysmetadata?view=az-ps-10

**Review by m-nash — APPROVED** [review]

**Pull request merged**